src/main/java/com/battleship/backend/models/ComputerPlayer.java

```diff
@@ -0,0 +1,117 @@
+        this.shipList = shipList;
+        this.playerBoard = boardRepository.getPlayerBoard();
+        this.computerBoard = boardRepository.getComputerBoard();
+        this.hitRequestValidator = hitRequestValidator;
+        this.placeShipsValidator = placeShipsValidator;
+
+        rand = new Random();
+        requestFactory = new RequestFactory();
+        name = "Computer Player";
+    }
+
+    public Request generateValidHitRequest(){
+        int randRow = rand.nextInt(playerBoard.getSize());
+        int randCol = rand.nextInt(playerBoard.getSize());
+
+        Request randHitRequest = requestFactory.generateHitRequest(randRow, randCol);
+
+        while(!hitRequestValidator.isValid(playerBoard, randHitRequest)){
+            randRow = rand.nextInt(playerBoard.getSize());
+            randCol = rand.nextInt(playerBoard.getSize());
+
+            randHitRequest = requestFactory.generateHitRequest(randRow, randCol);
+        }
+        return randHitRequest;
+    }
+
+    public Request generateValidPlaceRequest(Ship ship){
+        int randRow = rand.nextInt(computerBoard.getSize());
+        int randCol = rand.nextInt(computerBoard.getSize());
+
+        Request randPlaceRequest = requestFactory.generatePlaceRequest(ship, randRow, randCol);
+
+        while(!placeShipsValidator.isValid(computerBoard, randPlaceRequest)){
+            randRow = rand.nextInt(computerBoard.getSize());
+            randCol = rand.nextInt(computerBoard.getSize());
+
+            randPlaceRequest = requestFactory.generatePlaceRequest(ship, randRow, randCol);
+        }
+        return randPlaceRequest;
+    }
+
+    public void placeShip(Request placeRequest){
+        computerBoard.addShip(placeRequest.getShip(), placeRequest.getRow(), placeRequest.getCol());
+        removeShip(placeRequest.getShip());
+    }
+
+    public void placeAllShips(){
+        for(Ship ship : shipList){
+            Request placeRequest = generateValidPlaceRequest(ship);
+            placeShip(placeRequest);
+        }
+    }
+
+    private void removeShip(Ship shipToRemove){
+        Ship[] oldShipList = getShipList();
+        Ship[] newShipList = new Ship[oldShipList.length -  1];
+
+        for(int i = 0, j = 0; i < oldShipList.length; i++){
+            if(oldShipList[i] != shipToRemove){
+                newShipList[j] = oldShipList[i];
+                j++;
+            }
+        }
+        setShipList(newShipList);
+    }
+
+    public void fire(Request hitRequest){
+        playerBoard.hitSection(hitRequest.getRow(), hitRequest.getCol());
+    }
+
+    //Getters;
+
+    public BoardRepository getBoardRepository() {
+        return boardRepository;
+    }
+
+    public Ship[] getShipList() {
+        return shipList;
+    }
+
+    public Boardable getPlayerBoard() {
+        return playerBoard;
+    }
+
+    public Boardable getComputerBoard() {
+        return computerBoard;
+    }
+
+    public String getName() {
+        return name;
+    }
+
```

Comment:
Why do have these getters?  Is it just for tests?  In which case, you don't need it.

Reply:
Same comment for the setter.

Reply:
-Yeah those were basically for the tests. The setter is used though, in removeShip. Will remove the unnecessary ones.